### PR TITLE
Add upgrade tests for database level collation

### PR DESCRIPTION
**Pull request by ahmed-shameem**
### Description

This commit adds upgrade tests for database level collation. For any collatable default constraint, PG adds EXPLICIT collate clause for the default string (const node) iff the collation of the attribute is different from the collation of the datatype stored in pg_type catalog. This is the result of invocation of pg_get_constraintdef() function which finally calls get_const_expr(). To set the collation of the Const node, it calls get_const_collation() function where explicit COLLATE clause is added based on the aforementioned condition. 
Hence, when we create a database with non-default collation then the collatable columns get collated with the database level collation (if not mentioned explicitly). But we store the collation of the collatable datatypes as server level collation which fulfils the condition mentioned above. To solve this issue, we simply enclose the decompiled DEFAULT expression by PARENTHESIS to avoid invalid syntax.

Extension PR: https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2919
 
### Issues Resolved

Task: BABEL-5141
Authored-by: Shameem Ahmed [shmeeh@amazon.com](mailto:shmeeh@amazon.com)
Signed-off-by: Shameem Ahmed [shmeeh@amazon.com](mailto:shmeeh@amazon.com)
 
### Check List

- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the PostgreSQL license, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by rishabhtanwar29 — DISMISSED** [review]
LGTM

File: src/bin/pg_dump/dump_babel_utils.c [hunk 1]
```diff
@@ -503,8 +503,32 @@ fixTsqlDefaultExpr(Archive *fout, AttrDefInfo *attrDefInfo)
...
+	 * 	2. For the default expression
+	 * Eg: CREATE TABLE t1(a nvarchar(11) DEFAULT 'default') --> 
+	 * CREATE TABLE t1("a" "sys"."nvarchar" DEFAULT 'default'::"sys"."varchar" COLLATE <non_default_collation> COLLATE <non_default_collation>)
+	 * This is an invalid syntax, hence it will fail during restore.
+	 * If, attrDefInfo->adef_expr contains COLLATE clause, it means the DEFAULT
+	 * value is collatable. Hence we re-write it to:
+	 * CREATE TABLE t1("a" "sys"."nvarchar" DEFAULT ('default'::"sys"."varchar" COLLATE <non_default_collation>) COLLATE <non_default_collation>)
+	 * by adding parenthesis. 
+	 */
+	if (isBabelfishDb && strstr(source, " COLLATE ") != NULL)
```
Comment (rishabhtanwar29) [inline src/bin/pg_dump/dump_babel_utils.c]: Can this 'COLLATE' be in small case? In that case this comparison might fail.

Comment (ahmed-shameem) [reply src/bin/pg_dump/dump_babel_utils.c]: From the code I do not see that will be a problem, we always add explicit clause for Const node for default datatype like this `" COLLATE %s"`.

Comment (ahmed-shameem) [reply src/bin/pg_dump/dump_babel_utils.c]: Updated the logic in latest rev.

**Review by rishabhtanwar29 — APPROVED** [review]

**Review by rohit01010 — APPROVED** [review]

**Review by Deepesh125 — APPROVED** [review]

**Pull request merged**